classes/village/VillageUpgradeManager.php

```diff
@@ -0,0 +1,368 @@
+<?php
+
+require_once __DIR__ . "/VillageBuildingDto.php";
+require_once __DIR__ . "/VillageUpgradeDto.php";
+require_once __DIR__ . "/VillageUpgradeSetDto.php";
+require_once __DIR__ . "/VillageBuildingConfig.php";
+require_once __DIR__ . "/VillageUpgradeConfig.php";
+
+class VillageUpgradeManager {
+    const UPGRADE_TOGGLE_COOLDOWN_DAYS = 3;
+
+    /**
+     * @param System  $system
+     * @param int $village_id
+     * @return VillageBuildingDto[]
+     */
+    public static function getBuildingsForVillage(System $system, int $village_id): array {
+        $buildings = [];
+        $query = $system->db->query("
+            SELECT * FROM `village_buildings` WHERE `village_id` = {$village_id}
+        ");
+        $building_data = $system->db->fetch_all($query);
+        foreach ($building_data as $building) {
+            $buildings[$building['key']] = new VillageBuildingDto(
+                id: $building['id'],
+                key: $building['key'],
+                village_id: $building['village_id'],
+                tier: $building['tier'],
+                health: $building['health'],
+                max_health: VillageBuildingConfig::BUILDING_MAX_HEALTH[$building['key']][$building['tier']],
+                defense: $building['defense'],
+                status: $building['status'],
+                construction_progress: $building['construction_progress'],
+                construction_progress_required: $building['construction_progress_required'],
+                construction_progress_last_updated: $building['construction_progress_last_updated'],
+            );
+        }
+        return $buildings;
+    }
+
+    /**
+     * @param System  $system
+     * @param int $village_id
+     * @return VillageUpgradeDto[]
+     */
+    public static function getUpgradesForVillage(System $system, int $village_id): array {
+        $upgrades = [];
+        $query = $system->db->query("
+            SELECT * FROM `village_upgrades` WHERE `village_id` = {$village_id}
+        ");
+        $upgrade_data = $system->db->fetch_all($query);
+        foreach ($upgrade_data as $upgrade) {
+            $upgrades[$upgrade['key']] = new VillageUpgradeDto(
+                id: $upgrade['id'],
+                key: $upgrade['key'],
+                village_id: $upgrade['village_id'],
+                status: $upgrade['status'],
+                research_progress: $upgrade['research_progress'],
+                research_progress_required: $upgrade['research_progress_required'],
+                research_progress_last_updated: $upgrade['research_progress_last_updated'],
+            );
+        }
+        return $upgrades;
+    }
+
+    /**
+     * @param System  $system
+     * @param VillageUpgradeDto[] $upgrades
+     * @return array
+     */
+    public static function initializeEffectsForVillage(System $system, array $upgrades): array {
+        // initialize array with defaults
+        $effects = [
```

Comment:
Should this array of "what all effects are there" be defined in `VillageUpgradeConfig` instead and we can clone it in here and initialize all values to 0? Might reduce instances of adding a new effect and forgetting to update this list